### ISSUE TITLE
x2 speedup for LookupTable:updateOutput ? 

### ISSUE THREAD

**Issue by pengsun** [torch/cunn#263]
Hi,

Current `LookupTable:updateOutput` method calls the indexed copy function in cutorch, which is fast but still has room for speedup. 

By re-writing the cuda C with memory read/write coalescing, I found it is able to have a x2 to x2.3 speedup on big sequence batch. Even for small sequence batch the new impl is still slightly faster.

A snippet cu code for this is here: 
https://gist.github.com/pengsun/ac9d40f33deba1e36e18389110994fa2

And a simple timing lua code is here:
https://gist.github.com/pengsun/46ee5f67ec611252c0a58450f57768d5

Does it deserve a PR for this?

**Comment by soumith** [comment]
yes i think a PR would be nice. I'll let @colesbury confirm after he looks at your code.

**Comment by pengsun** [comment]
Hi @soumith, thanks! I just found that my code actually does the similar thing to a previous impl of indexSelect: 
https://github.com/torch/cutorch/commit/c626c99825e03b0c34ba793eea42880b8cf10754

But my code dosn't do thread block unrolling. Not sure if the optimization for contiguous data is still out there in current code, which is optimized for so many cases that I get lost in the details and fail to find it out. If the contiguous data with memory coalescing has been covered, then a new PR is unnecessary.
